### PR TITLE
Scipy 2022

### PR DESCRIPTION
**Pull request by mkshah605**

### CODE REVIEW

**Review by edrogers — CHANGES_REQUESTED** [review]
Thanks for this great tutorial. I really enjoyed it! Great job!

I wanted to offer a few notes related to what we discussed during the tutorial.

One discussion we spent a few minutes on was why, in notebook 4, it appears that land_mask and water_mask don't cover the entire rectangle of the image. Turns out, this is a matplotlib effect. I was able to zoom in and see that they matched with code like:
```python
plt.figure(3, figsize=(10,10))
plt.imshow(land_mask[np.ix_(range(2000,2100),range(2000,2100))])
plt.title("Houston land mask")
plt.colorbar()
```
```python
plt.figure(4, figsize=(10,10))
plt.imshow(water_mask[np.ix_(range(2000,2100),range(2000,2100))])
plt.title("Houston water mask")
plt.colorbar()
```

Feel free to include if you'd like.

Thanks again!

File: jupyter-notebooks/scipy-2022-workshop/3_Compute_NDWI.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/3_Compute_NDWI.ipynb:199]: I'd recommend setting the colour map to "blues", as is done in the `coastline_analysis` notebook. It really helps the reader visualize where the water resides in the maps.
```suggestion
    "fig = plt.imshow(ndwi)\n",
    "fig.set_cmap('Blues')\n",
```

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:168]: Same as above.
```suggestion
```

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:182]: Same as above.
```suggestion
```

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:206]: The `cv2` is more commonly referred to as OpenCV.
```suggestion
    "More info about morphological filters using OpenCV: https://docs.opencv.org/4.x/d9/d61/tutorial_py_morphological_ops.html"
```

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:215]: Add more context + comment on the line above, to be consistent with our coding standards.
```suggestion
    "# kernel size for empty pixels within the mask\n",
    "closing_kernel_size = 35,"
```

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:216]: Same as line above.
```suggestion
    "# kernel size for filled pixels outside of mask",
    "opening_kernel_size = 15"
```

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:238]: Doubling the accoladed memory for this array feels unnecessary. I'd recommend just either removing this line and replacing all instances of `water_mask_filtered` with `mask_closed_opened` OR on line 233 replacing `mask_closed_opened` with `water_mask_filtered`. That being said, it doesn't really matter. I'll yet you make the final call here. 
```suggestion
```

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:247]: Not needed, like above.
```suggestion
```

File: jupyter-notebooks/scipy-2022-workshop/coastline_analysis.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/coastline_analysis.ipynb:185]: This is the new recommended link from Matplotlib.
```suggestion
    "# see also: https://matplotlib.org/stable/gallery/color/named_colors.html\n",
```

File: jupyter-notebooks/scipy-2022-workshop/5_plotting_a_histogram.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/5_plotting_a_histogram.ipynb:143]: Since you're looking at blue-band data, why not set the colour to some shade of blue? Perhaps choose your favourite shade from https://matplotlib.org/stable/gallery/color/named_colors.html - my go-to is `color = royalblue`, but chef's choice!

File: jupyter-notebooks/scipy-2022-workshop/README.md
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/README.md:22]: Once we get this merged into main, we can get rid of the `git checkout` command and we can consolidate these `cd` commands :)
```suggestion
cd notebooks/jupyter-notebooks/scipy-2022-workshop/
```

**Comment by kevinlacaille** [comment]
Let's get rid of `AOI.png` and `pin.png`, they were just used to create `region.png`.

**Comment by kevinlacaille** [comment]
Great stuff @mkshah605 and thank you for your contributions @edrogers! Let's make these small changes, then I think it's good to go!

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (kevinlacaille) [reply jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:166]: Great finds!

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (kevinlacaille) [reply jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:180]: Thanks!

**Review by kevinlacaille — CHANGES_REQUESTED** [review]
Let's get these handful of tiny changes reviewed, then I think it's good to go!

File: jupyter-notebooks/scipy-2022-workshop/5_plotting_a_histogram.ipynb
Comment (mkshah605) [inline jupyter-notebooks/scipy-2022-workshop/5_plotting_a_histogram.ipynb:143]:
```suggestion
    "color = 'royalblue'\n",
```

File: jupyter-notebooks/scipy-2022-workshop/README.md
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/README.md:21]: Please add "```" here, as to end the `bash` code block.

File: jupyter-notebooks/scipy-2022-workshop/coastline_analysis.ipynb
Comment (mkshah605) [inline jupyter-notebooks/scipy-2022-workshop/coastline_analysis.ipynb:185]:
```suggestion
    "# see also: https://matplotlib.org/stable/tutorials/colors/colormaps.html\n",
```

**Review by kevinlacaille — CHANGES_REQUESTED** [review]

File: jupyter-notebooks/scipy-2022-workshop/3_Compute_NDWI.ipynb
Comment (kevinlacaille) [inline jupyter-notebooks/scipy-2022-workshop/3_Compute_NDWI.ipynb:200]:
```suggestion
    "plt.title(\"NDWI\")\n",
```

File: jupyter-notebooks/scipy-2022-workshop/README.md
Comment (mkshah605) [reply jupyter-notebooks/scipy-2022-workshop/README.md:21]: good catch!

Comment (mkshah605) [inline jupyter-notebooks/scipy-2022-workshop/README.md:21]:
```suggestion
```
```

File: jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb
Comment (mkshah605) [inline jupyter-notebooks/scipy-2022-workshop/4_masks_and_filters.ipynb:247]:
```suggestion
    "plt.imshow(mask_closed_opened)\n",
```

**Review by kevinlacaille — APPROVED** [review]
LGTM! 🎉

**Pull request merged**
